Sources/Plasma/PubUtilLib/plAvatar/plArmatureMod.cpp

```diff
@@ -859,6 +859,11 @@ void plArmatureMod::SpawnAt(int spawnNum, double time)
 
 void plArmatureMod::SetFollowerParticleSystemSO(plSceneObject *follower)
 {
+    if (!follower) {
+        fFollowerParticleSystemSO = follower;
```

Comment:
This doesn't seem to set up the key reference for the new follower (see line 873), which could cause the follower to be thrown away from under us.

Reply:
In this case, `follower` is null

Reply:
I'm still concerned about this. While it is true that `follower` is null, this seems like it's leaking the reference established on line 873. I do think it would be better to make it more obvious that `fFollowerParticleSystemSO` is being set to `nullptr`.

Reply:
It looks like there is a case (albeit perhaps unused in normal gameplay) where this method gets called with `nullptr`, and in theory the current code will crash.

If your preference to use the existing fFollowerParticleSystemSO to remove the reference (and maybe change `SetAttachedToAvatar(false)` on the particle system itself?) before setting it to null?

Reply:
I believe that's correct. We'll basically want to send another `plAttachMsg` with `plRefMsg::kOnRemove`, I think.

Reply:
I've updated this to send a detach message

Reply:
For future reference, I went through and verified that sending a `plAttachMsg` with the context `kOnRemove` will release the key reference. This happens in `plCoordinateInterface::IDetachChild()`.